src/Nethermind/Nethermind.Core/Crypto/PublicKey.cs

```diff
@@ -96,7 +97,17 @@ public override bool Equals(object? obj)
 
         public override int GetHashCode()
         {
-            return MemoryMarshal.Read<int>(Bytes);
+            byte[] bytes = Bytes;
+            long l0 = Unsafe.ReadUnaligned<long>(ref MemoryMarshal.GetArrayDataReference(bytes));
+            long l1 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long)));
+            long l2 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long) * 2));
+            long l3 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long) * 3));
+            long l4 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long) * 4));
+            long l5 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long) * 5));
+            long l6 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long) * 6));
+            long l7 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(bytes), sizeof(long) * 7));
+            l0 ^= l1 ^ l2 ^ l3 ^ l4 ^ l5 ^ l6 ^ l7;
```

Comment:
wouldn't it be faster to have only l0 and do ^= on each step? Would allocate less stack items.

Reply:
Jit sorts it out and turns it into a bunch of parallel xor loads
```
; Method Nethermind.Core.Crypto.PublicKey:GetHashCode():int:this
G_M000_IG01:                ;; offset=0000H

G_M000_IG02:                ;; offset=0000H
       488B4118             mov      rax, gword ptr [rcx+18H]
       488B5010             mov      rdx, qword ptr [rax+10H]
       48335018             xor      rdx, qword ptr [rax+18H]
       48335020             xor      rdx, qword ptr [rax+20H]
       48335028             xor      rdx, qword ptr [rax+28H]
       48335030             xor      rdx, qword ptr [rax+30H]
       48335038             xor      rdx, qword ptr [rax+38H]
       48335040             xor      rdx, qword ptr [rax+40H]
       48335048             xor      rdx, qword ptr [rax+48H]
       488BC2               mov      rax, rdx
       48C1F820             sar      rax, 32
       33C2                 xor      eax, edx

G_M000_IG03:                ;; offset=002DH
       C3                   ret   
```